kaggle_environments/core.py

```diff
@@ -643,21 +648,25 @@ def act(none_action=None):
         return structify({"act": act})
 
     def __get_shared_state(self, position):
-        if position == 0:
-            return self.state[0]
-        state = copy.deepcopy(self.state[position])
-
         # Note: state and schema are required to be in sync (apart from shared ones).
         def update_props(shared_state, state, schema_props):
             for k, prop in schema_props.items():
-                if get(prop, bool, path=["shared"], fallback=False):
+                # Hidden fields are tracked in the episode replay but are not provided to the agent at runtime
```

Comment:
This change adds the idea of hidden observation fields. To hide an item from agents you can set `"hidden": false` on the corresponding json schema entry. Here's an example of the attribute in action to hide bandit thresholds from agents: https://github.com/Kaggle/kaggle-environments/pull/116/files#diff-943836c947c0eb0017c5c1779d1ca8be9487cb6f47ee0df17a9a142fccc8a27dR57